### PR TITLE
Add redirect to ``about.readthedocs.com`` for logged out users

### PR DESCRIPTION
**Pull request by ericholscher**
This is the last step in our migration to a new marketing page,
showing it to logged out users on readthedocs.org.

Currently logged in users will hit a homepage at the root,
but this change makes that page redirect to the dashboard for community users.

Fixes https://github.com/readthedocs/readthedocs.org/issues/10571

### CODE REVIEW

**Review by stsewd — APPROVED** [review]

File: readthedocs/core/views/__init__.py [hunk 1]
```diff
@@ -45,15 +44,26 @@ class HomepageView(TemplateView):
...
-        context = super().get_context_data(**kwargs)
-        context['featured_list'] = Project.objects.filter(featured=True)
-        return context
+        # Redirect to user dashboard for logged in users
+        if request.user.is_authenticated:
+            return redirect("projects_dashboard")
+
+        # Redirect to ``about.`` in production
+        if not settings.DEBUG:
+            query_string = "?ref=dotorg-homepage"
```
Comment (agjohnson) [inline readthedocs/core/views/__init__.py:57]: I'm guessing we don't use this view on the commercial application, as we are already adding this to the requests on commercial. But if we do, this query string could be conditional based on the application instance.

**Review by agjohnson — APPROVED** [review]

**Comment by ericholscher** [comment]
@stsewd I expanded the logic removal here, so could use another review 👍

File: readthedocs/core/views/__init__.py (hunk 1 above)
Comment (ericholscher) [reply readthedocs/core/views/__init__.py:57]: Yea, we're overriding it. Could probably stop now though, if we wanted 🤔

**Review by stsewd — APPROVED** [review]
We can remove the override from .com once we are settled on this redirect on .org :+1:

**Comment by ericholscher** [comment]
Not going to merge this until we're 100% sure we're ready with the marketing website. Might hold off a week or two, but good to have it ready 👍

**Comment by ericholscher** [comment]
Our latest deploy is now out, so going to merge this so it's ready for next week 🎉

**Pull request merged**